solutions/cses-1662.mdx

```diff
@@ -80,5 +85,55 @@ int main() {
 	cout << ans << endl;
 }
 ```
+</CPPSection>
+	
+<JavaSection>
+	
+### Java Implementation
+	
+```java
+
+import java.util.*;
+import java.io.*;
+	
+// @Author Brad Ma
+
+public class subarrayDivisibility {
+	public static void main (String[] args) {
+		Kattio io = new Kattio();
+
+		int n = io.nextInt();
+
+		long[] M = new long[n];
+		long prefixSums = 0;
+		M[0] = 1;
+
+		for (int i = 0; i < n; i++) {
+			int a = io.nextInt();
+			prefixSums += a;
+			// remember to account for negative sums
+			M[((int) (prefixSums % n) + n) % n]++;
+		}
+
+		long answer = 0;
+		for (long x : M) {
+			/*
+			 *  calculating the # of pairs, this calculates the pairs without
+			 *  duplicates and reverse groups
+			 */
+
+			answer += x * (x - 1) / 2;
+		}
+
+		io.println(answer);
+		io.close();
+	}
+				      
+	//CodeSnip{Kattio}
+}
 
+```	
+</JavaSection>
+	
+</LanguageSection>
 </Spoiler>
```

Comment:
let's just remove the spoiler since most solutions don't have it
line 13 also needs to be deleted
```suggestion
```